### PR TITLE
fix: add missing typing for type arguments [type-arg] (TDE-266)

### PR DESCRIPTION
**Pull request by paulfouquet**
### Change Description:
Add the missing typing for type arguments. That is check by Mypy with error code type-arg

### Notes for Testing:
run `mypy .` should not display any error.

### CODE REVIEW

**Review by amfage — APPROVED** [review]

**Review by MDavidson17 — CHANGES_REQUESTED** [review]
Just a couple of unused imports

File: topo_processor/util/conversions.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import re
 from datetime import datetime
-from typing import Dict
+from typing import Any, Dict
```
Comment (MDavidson17) [inline topo_processor/util/conversions.py:3]: unused `Any' import?

File: topo_processor/util/aws_credentials.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import json
 import os
-from typing import Dict
+from typing import Any, Dict
```
Comment (MDavidson17) [inline topo_processor/util/aws_credentials.py:3]: Unused `Any` here?

File: topo_processor/stac/collection.py [hunk 3]
```diff
@@ -1,10 +1,10 @@
 from __future__ import annotations
 
 import os
-from datetime import datetime
+from datetime import date, datetime
```
Comment (MDavidson17) [inline topo_processor/stac/collection.py:4]: unused `Date' import?

File: topo_processor/util/conversions.py (hunk 1 above)
Comment (paulfouquet) [reply topo_processor/util/conversions.py]: Good spot. Fixed in https://github.com/linz/topo-processor/pull/512

File: topo_processor/util/aws_credentials.py (hunk 2 above)
Comment (paulfouquet) [reply topo_processor/util/aws_credentials.py:3]: Fixed in https://github.com/linz/topo-processor/pull/512

**Review by MDavidson17 — APPROVED** [review]
:+1:

**Pull request merged**